src/Interpreters/CacheLog.h

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#include <Core/NamesAndAliases.h>
+#include <Core/NamesAndTypes.h>
+#include <Core/Settings.h>
+#include <DataTypes/DataTypeArray.h>
+#include <DataTypes/DataTypeTuple.h>
+#include <Interpreters/SystemLog.h>
+#include <Interpreters/TransactionVersionMetadata.h>
+#include <base/logger_useful.h>
+
+namespace DB
+{
+struct CacheLogElement
+{
+    time_t event_time{};
+    String query_id;
+    String remote_file_path;
```

Comment:
I would also add here `Tuple file_segment_range;`
Because this information is important and can be found in `system.filesystem_cache`, but it is temporary and better to have it in this table too.

Reply:
What I mean here: It seems to me that it is not enough to have hits and misses counts. I want to know all file segment ranges per file and which file segments were used to read from cache, which to read from fs and download into cache, which were just bypassing cache.
So there will be many rows for the same cache path for same query id, each row for a separate file segment. And one more column: `read_type` (`CACHED`, `READ_FROM_FS_AND_DOWNLOADED_TO_CACHE`, `READ_FROM_FS_BYPASSING_CACHE`)
What do you think?

Reply:
> What I mean here: It seems to me that it is not enough to have hits and misses counts. I want to know all file segment ranges per file and which file segments were used to read from cache, which to read from fs and download into cache, which were just bypassing cache. So there will be many rows for the same cache path for same query id, each row for a separate file segment. And one more column: `read_type` (`READ_FROM_CACHE`, `READ_FROM_FS_AND_DOWNLOADED_TO_CACHE`, `READ_FROM_FS_BYPASSING_CACHE`) What do you think?

Yes, there is no doubt that the more detailed information recorded, the more conducive it is for users to analyze the behavior of a query. The minimum granularity of the current log is the access record of a query to a file. I also considered the record accurate to each segment before. However, this cost is the need to record more information in a query (which requires more time and space). If you think this cost can be ignored, I can implement the above functions. Maybe we can default that the function is not enabled. For users who need performance analysis, we can actively enable the log recording.

Reply:
Yes, let's have a more detailed records, under setting and not enabled by default.